nav2_behaviors/include/nav2_behaviors/behavior.hpp

```diff
@@ -48,26 +48,26 @@ enum class Status : int8_t
 using namespace std::chrono_literals;  //NOLINT
 
 /**
- * @class nav2_recoveries::Recovery
- * @brief An action server recovery base class implementing the action server and basic factory.
+ * @class nav2_behaviors::Behavior
+ * @brief An action server Behavior base class implementing the action server and basic factory.
  */
 template<typename ActionT>
-class Recovery : public nav2_core::Recovery
+class Behavior : public nav2_core::Behavior
```

Comment:
I like the idea of using this as an opportunity to clarify the "Recovery Plugin" vs "Recovery.hpp"-based Plugin distinction. I think we should add another word to distinguish between "behaviors" that use this time interface which also makes pluguns vs "behaviors" that are plugins

`TimedBehavior` `CycleBehavior` something like that?

Reply:
So, I renamed it for now but the timed behavior it is really a timed recovery because it has the vel_pub and stop robot func. 

We could factor out the vel_pub into the concrete implementation and provide a override for stopping the robot. `onActionTerimation` and `onActionSuccess`. 

However, I don't think that this is really important right now. 

I think we need to just add in some more behaviors and sort of get a feel for how things will look. 

I could not think of other behaviors we need besides assisted teleop but we could use some actions that are not recovery like.

Reply:
I think that is still useful for other behaviors, no? It could be a following behavior, that would need the velocity publisher. Or maybe its a "dance"  behavior. 

Do you think `Timed` is the right keyword? I think its better, but I'd be open to suggestions

Reply:
Yes, but we are restricted to behaviors that produce a velocity. 

Also, I don't even think it is a timed behavior interface. 
The timing is done in the concrete implementation by overriding `onRun` and `onCycleUpdate` which is left up to the implementer. 

Maybe `VelocityBehavior`?

Reply:
> Yes, but we are restricted to behaviors that produce a velocity.

Well that's a good point, that's what I had in mind. What kind of other behaviors are you thinking? I think of "behavior"s being spin, backup, get out of bad situations, assisted teleop, dance, etc which are all velocity-based things. There may be other types of behaviors, but the point of this server is to share assets like the costmap and TF buffer, which are most relevant (I think?) to these types of behaviors.

I was thinking the name more in line with trying to describe what this class is helping to implement: behaviors that run on a fixed cycle or timeline.

Reply:
Yeah, that makes sense. If we end up having thinking of more behaviors that need those assets but don't produce velocities we can always change it.

Reply:
OK - so `CycleBehavior`? 

What were you thinking though? I'd like to hear what you had in mind

Reply:
I don't think any of my ideas are really practical :)(Yell when the bot is close to an obstacle and some other silly stuff)! Also, sharing the costmap infers that the behaviors will produce a velocity.

I changed it to `TimedBehavior` if that works for you!